src/errors.rs

```diff
@@ -114,6 +114,61 @@ impl From<serde_json::Error> for PruneError {
     }
 }
 
+/// The errors that can be returned from [crate::sync::mount]
+#[derive(Debug)]
+pub enum MountError {
+    /// An unknown error occurred
+    Unknown(String),
+    /// Error while splitting the arguments
+    ShlexError,
+    /// The command failed to execute
+    CommandFailed(io::Error),
+    /// Invalid borg output found
+    InvalidBorgOutput(io::Error),
+    /// Error while deserializing output of borg
+    DeserializeError(serde_json::Error),
+    /// Borg was terminated by a signal
+    TerminatedBySignal,
+    /// Piping from stdout or stderr failed
+    PipeFailed,
```

Comment:
I think PipeFailed only occurs in `create_progress`, so we can just remove that here

Reply:
Other than that, I'm happy with the implementation

Reply:
Sounds good, removed